chore: synch with Substrate

wischli opened the pull request:
* Decrements toolchain to version used in Substrate for Polkadot v1.0.0 to v1.2.0
* Adds `2021` edition to fmt toml
* Adds license and description to root toml
* Extends gitignore

wischli committed:
chore: sync toolchain with Substrate

wischli committed:
chore: add 2021 edition to fmt toml

wischli committed:
chore: extend git ignore

wischli committed:
chore: improve root toml

lemunozm reviewed (commented):
I added `Cargo.lock` to the `.gitginore` because of this:

> If you’re building a non-end product, such as a rust library that other rust [packages](https://doc.rust-lang.org/cargo/appendix/glossary.html#package) will depend on, put Cargo.lock in your .gitignore

from https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

But never 100% sure about it

wischli committed:
fix: re-add Cargo.lock to gitignore

wischli commented:
> I added `Cargo.lock` to the `.gitginore` because of this:
> 
> > If you’re building a non-end product, such as a rust library that other rust [packages](https://doc.rust-lang.org/cargo/appendix/glossary.html#package) will depend on, put Cargo.lock in your .gitignore
> 
> from https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
> 
> But never 100% sure about it

Thanks for the clarification. I reverted that change, seems appropriate.

wischli committed:
ci: split sanity checks (#7)

* ci: split sanity checks

* ci: run clippy for all features

wischli committed:
ci: add names

wischli committed:
Merge branch 'main' into chore/rustfmt-toolchain

wischli committed:
Merge branch 'main' into chore/rustfmt-toolchain

lemunozm commented on rust-toolchain.toml:
```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "nightly-2023-08-24" # rustc 1.72.0 (2023-08-24)
+channel = "nightly-2023-05-22"  # rustc 1.71.0 8b4b20836
 components = [ "rustfmt", "clippy" ]
 targets = [ "wasm32-unknown-unknown" ]
 profile = "minimal"
```
Just wondering if being a library repo we really need this. I think not

@wischli

wischli replied:
You are probably right. I followed the path of Substrate which is not necessarily working with latest nightly versions. Feel free to remove the toolchain and adjust CI.

lemunozm replied:
Thanks for the info!

lemunozm replied:
After rethinking it, I think that although the users of this library do not need it, it makes sense for developers of the library itself (ours), so I think it is better to keep it.

Sorry for the noise!
